execution/scan/subquery.go

```diff
@@ -89,6 +89,11 @@ func (o *subqueryOperator) Next(ctx context.Context) ([]model.StepVector, error)
 		return nil, err
 	}
 
+	scalarArg := 0.0
+	if o.funcExpr.Func.Name == "quantile_over_time" {
+		scalarArg = o.funcExpr.Args[0].(*parser.StepInvariantExpr).Expr.(*parser.NumberLiteral).Val
```

Comment:
can we recursively unwrap here? I think we also do that in the matrix selector

Reply:
can we maybe put https://github.com/thanos-io/promql-engine/blob/8c1e6aea415471112118df6fb3d3bb61a1566883/storage/prometheus/scanners.go#L103

into "extscanners" module and import it here?